### ISSUE TITLE
addCFunction in 3rd party class

### ISSUE THREAD

**Issue by Mellnik** [vinniefalco/LuaBridge#285]
This library is just awesome but it seems that I'm overseeing something on this simple problem.

I have a class from a third party library which cannot be modified at all. I'm exposing this class with the usual **beginClass**.
Now I'd like to use **addCFunction** on this class but I cannot do that because it only accepts class member functions.
std::function also doesn't work here.

```
.addCFunction("MyCustomFunc", std::function<int(lua_State*)>([](lua_State* L) {
    luabridge::push(L, 1337);
    luabridge::push(L, 3.1415);
    return 2;
}))
```

How can I add a C function with lua_State and custom return values to my class which cannot be modified?

Thank you!

**Comment by kunitoki** [comment]
If you use addCFunction, you need to make sure you provide a lua_CFunction:
```cpp 
int myCFunction(lua_State*) { return 0; }

.addCFunction("MyCustomFunc", &myCFunction))
```

or eventually decay a lambda to a function pointer

```cpp
.addCFunction("MyCustomFunc", +[](lua_State*) -> int { return 0; })
```

**Comment by Mellnik** [comment]
Your first suggestion does not work. Because the class on which I want to do _addCFunction_ **cannot** be modified. It's a 3rd party class.
So when I do what you wrote it gives an error.

```
cannot convert argument 2 from 'int (__cdecl *)(lua_State *)' to 'int (__cdecl NameOfThe3dPartyClass::* )(lua_State *)'
```

The second suggestion with the lambda does not work as well for the same reason.

What luabridge is missing here is a specialization for addCFunction where the bound function is not part of the class.
Pretty much a "function member proxy" like for normal non-CFunctions as described here. http://vinniefalco.github.io/LuaBridge/Manual.html#s2.5

**Comment by kunitoki** [comment]
You are right, i was i thought you wanted to register a static method. can't you use a normal function ?

```cpp
.addFunction("MyCustomFunc", +[](MyClass* self, lua_State* L) { })
```

**Comment by Mellnik** [comment]
@kunitoki Nice, is it possible to use that piece of code in the "official" luabridge?

**Comment by kunitoki** [comment]
Yeah, not pretty but you can "hack it" somehow registering a static function to you class with addStaticFunction but call it with : instead of .

**Comment by dmitry-t** [comment]
Ok, let me implement this :)
